plugins/optimization-detective/class-od-url-metrics-group-collection.php

```diff
@@ -193,7 +193,7 @@ private function create_groups(): array {
 	 */
 	public function add_url_metric( OD_URL_Metric $new_url_metric ): void {
 		foreach ( $this->groups as $group ) {
-			if ( $group->is_viewport_width_in_range( $new_url_metric->get_viewport_width() ) ) {
+			if ( $group->is_viewport_width_in_range( $new_url_metric->viewport['width'] ) ) {
```

Comment:
In addition to what I said in https://github.com/WordPress/performance/pull/1492/files#r1752966800, this here is an example where I would even argue it was more ergonomical before. `get_viewport_width()` is more intuitive to use than `viewport['width']` since for the latter you have to know which keys are available within the viewport array. So that means it's not even always more ergonomical to use read-only properties.

Reply:
Yeah, this isn't necessarily more ergonomical. Nevertheless, the IDE should know that `$new_url_metric->viewport` is of type `ViewportRect` so it should autocomplete the keys:

![image](https://github.com/user-attachments/assets/6719b92f-79b6-42da-ac6f-9c9e8d207857)

Reply:
Undone in 85a03a96